File: integraupt-web/Integraupt-Backend/BackendSanciones/src/main/java/com/Sanciones/Controller/UsuarioBusquedaController.java
package com.Sanciones.controller;

import com.Sanciones.dto.usuario.UsuarioBusquedaResponse;
import com.Sanciones.service.UsuarioBusquedaService;
import org.springframework.util.StringUtils;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping("/api/usuarios")
@CrossOrigin(origins = "*")
public class UsuarioBusquedaController {

    private final UsuarioBusquedaService usuarioBusquedaService;

    public UsuarioBusquedaController(UsuarioBusquedaService usuarioBusquedaService) {
        this.usuarioBusquedaService = usuarioBusquedaService;
    }

    @GetMapping("/busqueda")
    public List<UsuarioBusquedaResponse> buscarUsuarios(@RequestParam("tipoUsuario") String tipoUsuario,
                                                        @RequestParam(value = "rol", required = false) String rol,
                                                        @RequestParam(value = "query", required = false) String query,
                                                        @RequestParam(value = "facultadId", required = false) Long facultadId,
                                                        @RequestParam(value = "escuelaId", required = false) Long escuelaId,
                                                        @RequestParam(value = "escuelaContextoId", required = false) Long escuelaContextoId,
                                                        @RequestParam(value = "limit", required = false) Integer limit) {
        String termino = StringUtils.hasText(query) ? query.trim() : null;
        return usuarioBusquedaService.buscarUsuarios(tipoUsuario, rol, termino, facultadId, escuelaId, escuelaContextoId, limit);
    }
}

File: Integraupt-Backend/servicio-sanciones/app/Http/Controllers/UsuarioBusquedaController.php
<?php

namespace App\Http\Controllers;

use App\Services\UsuarioBusquedaService;
use Illuminate\Http\JsonResponse;
use Illuminate\Http\Request;

class UsuarioBusquedaController extends Controller
{
    public function __construct(private UsuarioBusquedaService $usuarioBusquedaService)
    {
    }

    public function buscarUsuarios(Request $request): JsonResponse
    {
        try {
            $resultados = $this->usuarioBusquedaService->buscarUsuarios(
                $request->query('tipoUsuario'),
                $request->query('rol'),
                $request->query('query'),
                $request->query('facultadId') !== null ? (int) $request->query('facultadId') : null,
                $request->query('escuelaId') !== null ? (int) $request->query('escuelaId') : null,
                $request->query('escuelaContextoId') !== null ? (int) $request->query('escuelaContextoId') : null,
                $request->query('limit') !== null ? (int) $request->query('limit') : null,
            );

            return response()->json($resultados);
        } catch (\InvalidArgumentException $e) {
            return response()->json($e->getMessage(), 400);
        }
    }
}
